### DISCUSSION TITLE
Readonly CTkOptionMenu

### DISCUSSION THREAD

**Discussion by sathyarajshetigar** [TomSchimansky/CustomTkinter#1000] [Q&A] [answered]
Hi, Now CTkOptionMenu entry can be modified by clicking and typing in anything. Can we force users to select only what's fed initially when creating the optionMenu?

**Comment by Akascape** [comment, marked as answer]
@sathyarajshetigar CTkOptionMenu has no entry field. 
CTkComboBox and CTkOptionMenu are both different widgets. 
CTkOptionMenu has no entry field, but CTkComboBox has a entry field which can be disabled by the readonly parameter.

Here is an example for the combobox with readonly mode:
```python
combobox = customtkinter.CTkComboBox(root, state="readonly", values=["A", "B", "C", "D"])
combobox._entry.configure(readonlybackground=combobox._apply_appearance_mode(combobox._fg_color))
combobox.pack()
```

**Reply by sathyarajshetigar** [reply]
Awesome.
![image](https://user-images.githubusercontent.com/42057975/211277993-0d7a7c2f-f588-49c7-a40f-13591e54591a.png)
How to change that white color to something else? its that disabled state. tried setting all possible color parameters but white color does not get changed.

**Reply by Akascape** [reply]
@sathyarajshetigar You have to add this if you are using readonly mode:
```python
combobox._entry.configure(readonlybackground=combobox._apply_appearance_mode(combobox._fg_color))
```
I have mentioned this in the example. 
(This is actually a bug)

Make sure that you are using the latest version of customtkinter.

**Reply by sathyarajshetigar** [reply]
Yes. somehow I missed that. Sorry for the trouble. Now it look alright.
